From 8e4363702a79166ad20ceb0136a2d8bfb18d4e91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sebastian=20Dr=C3=B6ge?= <sebastian.droege@collabora.co.uk>
Date: Mon, 12 Mar 2012 14:30:02 +0100
Subject: [PATCH 5/8] Blacklist the bamf GIO module

I uses Ubuntu specific GLib API on Ubuntu... and thus crashes
everything that ever tries to use GIO.
---
 gio/giomodule.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gio/giomodule.c b/gio/giomodule.c
index 06bae56..4b3dccf 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -352,6 +352,13 @@ is_valid_module_name (const gchar        *basename,
 {
   gboolean result;
 
+  /* bamf module that uses Ubuntu specific GLib
+   * API on Ubuntu... and thus crashes everything
+   * that ever tries to use GIO.
+   */
+  if (strcmp (basename, "libgiobamf.so") == 0)
+    return FALSE;
+
 #if !defined(G_OS_WIN32) && !defined(G_WITH_CYGWIN)
   if (!g_str_has_prefix (basename, "lib") ||
       !g_str_has_suffix (basename, ".so"))
-- 
2.1.0

